apiserver/facades/agent/sshconnreq/facade.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshconnreq
+
+import (
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/watcher"
+)
+
+// Backend provides required state for the Facade.
+type Backend interface {
+	GetSSHConnRequest(docID string) (state.SSHConnRequest, error)
+	WatchSSHConnRequest(machineId string) state.StringsWatcher
+}
+
+// Facade allows model config manager clients to watch controller config changes and fetch controller config.
+type Facade struct {
+	resources facade.Resources
+
+	backend Backend
+}
+
+func NewFacade(ctx facade.Context, backend Backend) *Facade {
+	return &Facade{
+		resources: ctx.Resources(),
+		backend:   backend,
+	}
+}
+
+// GetSSHConnRequest returns a ssh connection request by its document ID.
+func (f *Facade) GetSSHConnRequest(arg string) (params.SSHConnRequestResult, error) {
+	result := params.SSHConnRequestResult{}
+	connReq, err := f.backend.GetSSHConnRequest(arg)
+	if err != nil {
+		return result, err
+	}
+	result.SSHConnRequest = params.SSHConnRequest(connReq)
```

Comment:
i would rather see an explicit conversion - if we ever change either of these types, this won't compile